src/engine/IdTable.h

```diff
@@ -386,8 +393,13 @@ class IdTableImpl<0> {
     Row& operator*() { return _rowView; }
     const Row& operator*() const { return _rowView; }
 
-    Id& operator[](size_t i) { return *(_data + _row * _cols + i); }
-    const Id& operator[](size_t i) const { return *(_data + _row * _cols + i); }
+    // access the element the is i steps ahead
+    // we need to construct now rows for this which should not be too expensive
+    Row operator[](size_t i) { return Row(_data + (_row + i) * _cols, _cols); }
```

Comment:
The full row would be at `_data + _row * _cols` so I'm not sure what this returns.

Reply:
Concerning the Math:: see above.
We cannot return a reference since our idTable does not actually store Rows.
But changing the row also changes the IdTable in the expected way so we behave like a reference.

Reply:
Ahh, I missed that this is actually in `Row::iterator::operator[]()` not in `Row::operator[]()`  now this makes sense